### PR TITLE
Add RoBERTa example

### PR DESCRIPTION
**Pull request by tocean**
Add RoBERTa example

### CODE REVIEW

File: RoBERTa/run.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        --vocab-bpe gpt2_bpe/vocab.bpe \
+        --inputs wikitext-103-raw/wiki.${SPLIT}.raw \
+        --outputs wikitext-103-raw/wiki.${SPLIT}.bpe \
+        --keep-empty \
+        --workers 60; \
+done
+
+
+wget -O gpt2_bpe/dict.txt https://dl.fbaipublicfiles.com/fairseq/gpt2_bpe/dict.txt
+fairseq-preprocess \
```
Comment (wkcn) [inline RoBERTa/run.sh:31]: It seems to be a data process script, rather than training script.

Comment (tocean) [reply RoBERTa/run.sh]: Yes. It has been fixed.

File: RoBERTa/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    ├── train.bin
+    ├── train.idx
+    ├── valid.bin
+    └── valid.idx
+```
+
+If you use your own data, don't forget to change the variable DATA_PATH to the data folder in launch script.
+
+## Apply patch to fairseq
+We made a few changes to the official [failseq](https://github.com/facebookresearch/fairseq) and packaged it into a patch. You need to apply this patch to third_party/fairseq.
```
[the remark this answers is not in the document]
Comment (tocean) [reply RoBERTa/README.md]: Fixed.

**Review by wkcn — APPROVED** [review]
LGTM. Thanks Yuxiang!

File: RoBERTa/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# This is an example of RoBERTa using MS-AMP
+This example demonstrates how to MS-AMP in [RoBERTa](https://github.com/facebookresearch/fairseq/blob/main/examples/roberta/README.md).
```
Comment (guoshzhao) [inline RoBERTa/README.md:2]: `how to MS-AMP` => `how to use MS-AMP`

File: RoBERTa/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+git apply ../../RoBERTa/fairseq.patch
+```
+
+## Install failseq
+You need to install fairseq before training RoBERTa. It is recommended to use venv for virtual environments, but it is not strictly necessary.
+```
+pip install --no-build-isolation -v -e .
+cd -
+```
+You can verify if the installation of fairseq by executing `python -c "import fairseq; print(fairseq.__version__)"`.
```
Comment (guoshzhao) [inline RoBERTa/README.md:41]: `You can verify if the installation of fairseq by executing` =>
`You can verify if the installation of fairseq is succesful by executing`

File: RoBERTa/run.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        $fairseq_train \
+        --config-dir ../third_party/fairseq/examples/roberta/config/pretraining \
+        --config-name base  \
+        task.data=$DATA_PATH \
+        checkpoint.save_dir=$SAVE_PATH \
+        dataset.skip_invalid_size_inputs_valid_test=True \
+        dataset.batch_size=64 \
+        optimization.update_freq=[8] \
+        common.fp16=False \
+        common.amp=True \
```
Comment (guoshzhao) [inline RoBERTa/run.sh:57]: Both `common.amp` and `common.msamp` are `True`?

**Review by guoshzhao — APPROVED** [review]

File: RoBERTa/README.md (hunk 3 above)
Comment (tocean) [reply RoBERTa/README.md]: Fixed.

File: RoBERTa/README.md (hunk 4 above)
Comment (tocean) [reply RoBERTa/README.md]: Fixed.

File: RoBERTa/run.sh (hunk 5 above)
Comment (tocean) [reply RoBERTa/run.sh:57]: Yes. The baseline is amp. We need to enable amp when using ms-amp to get a comparable performance result.

**Pull request merged**